scripts/create_repository.py

```diff
@@ -0,0 +1,198 @@
+    coordinates = get_maven_coordinates(replaced)
+    checksum = get_artifact_checksum(replaced)
+    direct_dependencies = get_json_dependencies(replaced)
+    resolved_artifacts.append(ResolvedArtifact(coordinates, checksum, direct_dependencies))
+  return resolved_artifacts
+
+def resolve_artifacts_with_checksums_and_direct_dependencies(root_artifacts) -> List[ResolvedArtifact]:
+  command = f'cs resolve {' '.join(root_artifacts)}'
+  output = subprocess.run(command, capture_output=True, text=True, shell=True).stdout.splitlines()
+  return map_to_resolved_artifacts(output)
+
+def to_rules_scala_compatible_dict(artifacts, version) -> Dict[str, Dict]:
+  temp = {}
+
+  for a in artifacts:
+    label = get_label(a.coordinates).replace('scala3_', 'scala_').replace('scala_tasty_core', 'scala_scala_tasty_core')
+    deps = [f'@{get_label(dep)}_2' if "scala3-library_3" in a.coordinates.artifact else f'@{get_label(dep)}' for dep in a.direct_dependencies]
+
+    temp[label] = {
+        "artifact": f"{a.coordinates.coordinate}",
+        "sha256": f"{a.checksum}",
+    } if not deps else {
+        "artifact": f"{a.coordinates.coordinate}",
+        "sha256": f"{a.checksum}",
+        "deps:": deps,
+      }
+
+  return temp
+
+def is_that_trailing_coma(content, char, indice) -> bool:
+  return content[indice] == char and content[indice+1] != ',' and content[indice+1] != ':' and content[indice+1] != '@' and not content[indice+1].isalnum()
+
+def get_with_trailing_commas(content) -> str:
+  copied = copy.deepcopy(content)
+  content_length = len(copied)
+  i = 0
+
+  while i < content_length - 1:
+    if is_that_trailing_coma(copied, '"', i):
+      copied = copied[:i] + '",' + copied[i + 1:]
+      content_length = content_length + 1
+      i = i+2
+    elif is_that_trailing_coma(copied, ']', i):
+      copied = copied[:i] + '],' + copied[i + 1:]
+      content_length = content_length + 1
+      i = i+2
+    elif is_that_trailing_coma(copied, '}', i):
+      copied = copied[:i] + '},' + copied[i + 1:]
+      content_length = content_length + 1
+      i = i+2
+    else:
+      i = i+1
+
+  return copied
+
+def write_to_file(artifact_dict, version, file):
+  with file.open('w') as data:
+    data.write(f'scala_version = "{version}"\n')
+    data.write('\nartifacts = ')
+    data.write(f'{get_with_trailing_commas(json.dumps(artifact_dict, indent=4).replace('true', 'True').replace('false', 'False'))}\n')
+
+def create_file(version):
+  path = os.getcwd().replace('/scripts', '/third_party/repositories')
+  file = Path(f'{path}/{'scala_' + "_".join(version.split(".")[:2]) + '.bzl'}')
+
+  if not file.exists():
+    file_to_copy = Path(sorted(glob.glob(f'{path}/*.bzl'))[-1])
+    with file.open('w+') as data, file_to_copy.open('r') as data_to_copy:
+      for line in data_to_copy:
+        data.write(line)
+
+  with file.open('r+') as data:
+    excluded_artifacts = ["org.scala-lang.modules:scala-parser-combinators_2.11:1.0.4"]
```

Comment:
There are artifacts versions that cause CI-check failures after update and exceptions in logs.